example/errormsg/errormsg_test.go

```diff
@@ -0,0 +1,159 @@
+package errormsg
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/goccmack/gocc/example/errormsg/errors"
+	"github.com/goccmack/gocc/example/errormsg/lexer"
+	"github.com/goccmack/gocc/example/errormsg/parser"
+	"github.com/goccmack/gocc/example/errormsg/token"
+)
+
+func assertEqual(t *testing.T, lhs interface{}, rhs interface{}) {
+	t.Helper()
+	if lhs != rhs {
+		t.Fatalf("mismatch: expected %v, got %v", lhs, rhs)
+	}
+}
+
+func parse(t *testing.T, code string, expectPass bool) {
+	t.Helper()
+
+	l := lexer.NewLexer([]byte(code))
+	p := parser.NewParser()
+	_, err := p.Parse(l)
+
+	// Match err to expectPass (nil = passed, !nil = failed), and log if not nill
+	if err == nil {
+		if !expectPass {
+			t.Fatal("test should have failed, got nil error instead")
+		}
+	} else {
+		if expectPass && err != nil {
+			t.Fatalf("test should have passed, got error instead: %s", err.Error())
+		}
+		t.Log(err.Error())
+	}
+}
+
+func TestParsedErrors(t *testing.T) {
+	t.Run("pass", func(t *testing.T) {
+		candidates := []string{
+			"var abcd = 123;",
+			"  var  _  :=  abcd  ;",
+			"var a  = 1.23 ;",
+			"var x;",
+		}
+		for _, candidate := range candidates {
+			candidate := candidate
+			t.Run(candidate, func(t *testing.T) {
+				parse(t, candidate, true)
+			})
+		}
+	})
+
+	t.Run("EOF", func(t *testing.T) {
```

Comment:
Nit, these could just be separate tests instead of subtests

Reply:
I was inclined to let vscode/goland do their thing and make these just a testcase loop.